server/text_generation_server/cli.py

```diff
@@ -186,8 +186,11 @@ def download_weights(
             class_ = getattr(transformers, architecture)
 
             # Name for this varible depends on transformers version.
-            discard_names = getattr(class_, "_tied_weights_keys", [])
-            discard_names.extend(getattr(class_, "_keys_to_ignore_on_load_missing", []))
+            discard_names = []
+            if getattr(class_, "_tied_weights_keys", []):
+                discard_names.extend(getattr(class_, "_tied_weights_keys", []))
+            if getattr(class_, "_keys_to_ignore_on_load_missing", []):
+                discard_names.extend(getattr(class_, "_keys_to_ignore_on_load_missing", []))
```

Comment:
Actually I think we should just discard `_keys_to_ignore_on_load_missing`.

This should always be lists if defined, but indeed transformers recently switched everything to `_tied_weights_keys`.